src/main/java/com/example/bak/auth/infra/jwt/TokenProvider.java

```diff
@@ -0,0 +1,33 @@
+package com.example.bak.auth.infra.jwt;
+
+import com.example.bak.auth.infra.jwt.persistence.TokenType;
+import com.example.bak.auth.infra.jwt.persistence.AccessTokenProperties;
+import com.example.bak.auth.infra.jwt.persistence.JwtProperties;
+import com.example.bak.auth.infra.jwt.persistence.RefreshTokenProperties;
+import com.example.bak.auth.infra.jwt.persistence.Token;
+
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.stereotype.Component;
+
+import javax.crypto.SecretKey;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+public class TokenProvider {
+    private final Map<TokenType, Token> tokens = new HashMap<>();
+
+    public TokenProvider(JwtProperties tokenProperties) {
+        tokens.put(TokenType.ACCESS, AccessTokenProperties.from(tokenProperties.getAccess()));
+        tokens.put(TokenType.REFRESH, RefreshTokenProperties.from(tokenProperties.getRefresh()));
+    }
+
+    public SecretKey getSecretKey(TokenType tokenType) {
+        return Keys.hmacShaKeyFor(Decoders.BASE64.decode(tokens.get(tokenType).secretKey()));
+    }
```

Comment:
```suggestion
    public SecretKey getSecretKey(TokenType tokenType) {
        타입 시크릿 = Decoders.BASE64.decode(tokens.get(tokenType).secretKey());
        return Keys.hmacShaKeyFor(분리한 내용);
    }
```

이런 형태로 수정을 하는게 추후에 알아보기 더 편할 것 같은데 @Sunja-An 님은 어떻게 생각하세요?

Reply:
해당 의견에 대해서는 저도 동의를 하는 부분입니다!
1줄 코드로 적는게 멋있다는 강박관념이 아직 남아 있는 것 같네요 ㅎㅎ..😔